File: Java/LeetCode/Array/FindNumEvenDigits.py

class Solution:
    def findNumbers(self, nums: list[int]) -> int:
        def getNumOfDigits(num: int):
            digits = 0
            while (num > 0):
                digits += 1
                num = num // 10
            return digits
        count = 0
        for num in nums:
            if (getNumOfDigits(num) % 2 == 0):
                count += 1 

        return count

def main():
    sol = Solution ()
    nums = [12, 345, 2, 6, 7896]
    print(sol.findNumbers(nums))
    nums = [555,901,482,1771]
    print(sol.findNumbers(nums))

if __name__ == '__main__':
    main()

File: Java/LeetCode/Array/FindNumEvenDigits.java
// LC 1295 -- Array

public class FindNumEvenDigits {
    public int findNumbers(int[] nums) {
        int result = 0;
        for (int num: nums) {
            if (getNumOfDigits(num) % 2 == 0) {
                result++;
            }
        } 
        return result;
    }

    private int getNumOfDigits(int num) {
        int digits = 0;
        while (num > 0) {
            digits++;
            num = (int) (num / 10);
        }
        return digits;
    }
}
